inst/tests/tests.Rraw

```diff
@@ -1350,8 +1351,8 @@ test(457, DT[DT2, list(val, val2), nomatch=0, by=.EACHI], data.table(id=c(1L,2L,
 # Test bysameorder edge cases, #1631
 DT = data.table(a=1:3,v=4:9,key="a")
 test(458, DT[,sum(v),by=list(a%%2L)], data.table(a=c(1L,0L),V1=c(26L,13L)))
-test(459, DT[, list(sum(v)), list(ifelse(a == 2, NA, 1L))], data.table(ifelse=c(1L,NA_integer_),V1=c(26L,13L)))
-test(460, DT[, list(sum(v)), list(ifelse(a == 2, 1, NA))], data.table(ifelse=c(NA_real_,1),V1=c(26L,13L)))
+test(459, DT[, list(sum(v)), list(fifelse(a == 2L, NA_integer_, 1L))], data.table(fifelse=c(1L,NA_integer_),V1=c(26L,13L)))
```

Comment:
I think we should leave `ifelse` and not replace it with `fifelse`, better to test against base R than our functions.

Reply:
we do have a test (2085.33) matching `fifelse` to `ifelse`, I figured that is good enough. We could also add a few more tests of consistency of `fifelse` & `ifelse`...

Reply:
Testing `fifelse` is a different thing. We do test grouping by same order, not `fifelse` really. I is fine to use `fifelse` here, but we should not remove every `ifelse` in favour of `fifelse`. I would remove `ifelse` checks from linter.